src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,75 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+interface Props {
+  query: string;
+  onQueryChange: (str: string) => void;
+  showActive: () => void;
+  shawAllTodos: () => void;
+  showCompleted: () => void;
+}
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+export const TodoFilter: React.FC<Props> = ({
+  query,
+  onQueryChange,
+  showActive,
+  shawAllTodos,
+  showCompleted,
+}) => {
+  const handleFilterChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedValue = event.target.value;
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+    if (selectedValue === 'active') {
+      showActive();
+    } else if (selectedValue === 'all') {
+      shawAllTodos();
+    } else if (selectedValue === 'completed') {
+      showCompleted();
+    }
```

Comment:
```suggestion
   setTodosFilter(event.target.value);
```
pass setTodosFilter in this component and just set event.target.value.
no need to make such huge if else chain